tests/test_pxssh.py

```diff
@@ -274,5 +274,29 @@ def test_failed_custom_ssh_cmd(self):
         else:
             assert False, 'should have raised exception, pxssh.ExceptionPxssh'
 
+    def test_login_bash(self):
+        ssh = pxssh.pxssh()
+        result = ssh.login('server bash', 'me', password='s3cret')
+        ssh.sendline('ping')
+        ssh.expect('pong', timeout=10)
+        assert ssh.prompt(timeout=10)
+        ssh.logout()
+
+    def test_login_zsh(self):
+        ssh = pxssh.pxssh()
+        result = ssh.login('server zsh', 'me', password='s3cret')
+        ssh.sendline('ping')
+        ssh.expect('pong', timeout=10)
+        assert ssh.prompt(timeout=10)
+        ssh.logout()
+
+    def test_login_tcsh(self):
+        ssh = pxssh.pxssh()
+        result = ssh.login('server tcsh', 'me', password='s3cret')
+        ssh.sendline('ping')
+        ssh.expect('pong', timeout=10)
+        assert ssh.prompt(timeout=10)
+        ssh.logout()
```

Comment:
I wish `@pytest.mark.parametrize` worked in a `unittest`-style test class :(

Reply:
It does but not in the way you'd think.
https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests
https://github.com/Red-M/RedSSH/blob/a7d2cf12ebb5f9a4955509c55264212aff14bf2f/tests/test_auth.py#L16

Reply:
True, though it doesn't seem to then have the nice don't-stop-at-first-failure and list-parameter-with-testcase-name behaviour. Probably not worth it with only three shells being tested, but if more are added later might be worth considering.